config/config.yml

```diff
@@ -13,27 +16,28 @@ providers:
 permissions:
   # see full format details in 'permissions.cfg'
   # for each definition:
-  #   can refer to users and groups that will be created dynamically as needed with corresponding information
-  #   plain user or group name not matched against below entry will create it with defaults for other fields
+  #   can refer to below 'users' and 'groups' that will be created dynamically as needed with corresponding information
+  #   user or group name not matched against any extended definition will be created with defaults for other fields
 
 # extended definitions for users to create
 # default values will be used for missing optional fields
 #   not providing an entry referenced somewhere else (e.g.: in permissions)
 #   will default to only using the user name and defaults for every other field
 users:
   - username: username  # required if entry provided
-    password:           # optional
-    email:              # optional
-    group: <groupname>  # optional (either one of the groups, or plain string to generate it with defaults)
+    password:           # optional (default: random password of size MAGPIE_PASSWORD_MIN_LENGTH)
```

Comment:
Just wondering why this is optional as if a random password is used, nobody can access it and thus the whole account is useless.

Reply:
Since the admin can override, it is not critical to be missing the information. 
Also, without this default, there is no way to simply have the username string everywhere else that `user` is specified. 

Having the default allows to quickly define a few users for testing purposes. 
Obviously, for users that would be used for actual operations, the password would have to be provided in the definition.